### PR TITLE
CSM-7414: TerraformChanges for - TGW- multicast domain associations

### PR DESCRIPTION
**Pull request by pvadapalli-uptycs**
Adding Read Permission for "GetTransitGatewayMulticastDomainAssociations" API. 
Required for table: `aws_ec2_transit_gw_multicast_domain_association`
Related Ticket: https://uptycsjira.atlassian.net/browse/CSM-7207

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -42,6 +42,7 @@ resource "aws_iam_role" "role" {
             "ec2:DescribeAccountAttributes",
             "ec2:GetEbsEncryptionByDefault",
             "ec2:SearchTransitGatewayRoutes",
+            "ec2:GetTransitGatewayMulticastDomainAssociations",
```
Comment (djuneja-uptycs) [inline main.tf:45]: please keep them in lexicographical manner

Comment (pvadapalli-uptycs) [reply main.tf:45]: Ok

Comment (pvadapalli-uptycs) [reply main.tf]: Fixed.

**Review by djuneja-uptycs — APPROVED** [review]
@pvadapalli-uptycs make sure this is release noted

**Pull request merged**